pkg/controller/gkenetworkparamset/gkenetworkparamset_controller.go

```diff
@@ -255,8 +266,7 @@ func (c *Controller) reconcile(ctx context.Context, key string) error {
 
 	params := originalParams.DeepCopy()
 
-	err = c.syncGNP(ctx, params)
-
+	err = c.syncGNP(ctx, params, originalParams)
```

Comment:
It looks like you have done this because you want to modify 'params' within this function. A better pattern is to only pass 'originalParams' to the function, create a deep copy within that function if required and return the modified params from the function.

Reply:
Change to only pass the `params`, please review again.